tests/functional/test_config_upgrader.py

```diff
@@ -221,9 +221,13 @@ def test_upgrader_where_host_is_function(tmp_path, caplog):
     caplog.set_level(logging.DEBUG)
```

Comment:
I don't understand why this file has been moved to `functional/`?

Reply:
I think that at one point I began to believe that this was more functional than unit, but it isn't.

Reply:
So what aspect of this test is functional?

The "line in the sand" that I have in my head is:

* Integration - Involves the interaction of more than one significant piece of logic.
* Functional - Involves interaction with the shell, browser or other systems.

Reply:
I think this has been moved back again?